common/middleware.py

```diff
@@ -9,40 +9,49 @@ def __init__(self, get_response=None):
 
     def __call__(self, request):
         response = self.get_response(request)
+
         try:
             content = json.loads(response.content.decode())
         except json.JSONDecodeError:
             return response
 
-        if 'data' in content and request.method == 'GET':
+        if 'data' in content:
+            if content['data'].get('__schema'):
+                return response
             flattened_content = {}
             index = 0
             for resource in list(content['data'].values()):
                 # Use case for retrieval of all records
-                if (resource and 'edges' in resource
-                        and isinstance(resource['edges'], list)
-                        and 'node' in resource['edges'][0]):
+                key = list(content['data'].keys())[index]
+                if resource and 'edges' in resource and isinstance(resource['edges'], list) \
```

Comment:
Instead of splitting with `\`, I'm thinking you should break the statement. @kola-er what do you think?